core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1949,6 +1951,19 @@ class KafkaConfig(val props: java.util.Map[_, _], doLog: Boolean, dynamicConfigO
       )
     }
 
+    val voterIds: Set[Integer] = if (usesSelfManagedQuorum) RaftConfig.parseVoterConnections(quorumVoters).asScala.keySet.toSet else Set.empty
+    if (voterIds.nonEmpty) {
+      if (processRoles.contains(ControllerRole)) {
+        // Ensure that controllers use their node.id as a voter in controller.quorum.voters 
+        require(voterIds.contains(nodeId), s"If ${KafkaConfig.ProcessRolesProp} contains the 'controller' role, the node id $nodeId must be included in the set of voters ${RaftConfig.QUORUM_VOTERS_CONFIG}=$voterIds")
```

Comment:
In the interest of making it consistent with all the `KafkaConfig` properties shouldn't we have an variable for `RaftConfig.QUORUM_VOTERS_CONFIG`? Maybe `KafkaConfig.QuorumVotersProp`. This comment applies to a few places in this change.